diff -up wireshark-0.99.7-SVN-23507/Makefile.in.pie wireshark-0.99.7-SVN-23507/Makefile.in
--- wireshark-0.99.7-SVN-23507/Makefile.in.pie	2007-11-20 05:34:36.000000000 -0500
+++ wireshark-0.99.7-SVN-23507/Makefile.in	2007-11-20 15:00:25.000000000 -0500
@@ -1004,7 +1004,7 @@ noinst_HEADERS = \
 wireshark_optional_objects = @GETOPT_O@ @STRERROR_O@ \
 	@STRCASECMP_O@ @STRNCASECMP_O@ @MKSTEMP_O@ @STRPTIME_O@
 
-@ENABLE_STATIC_FALSE@wireshark_LDFLAGS = -export-dynamic
+@ENABLE_STATIC_FALSE@wireshark_LDFLAGS = -export-dynamic -pie
 @ENABLE_STATIC_TRUE@wireshark_LDFLAGS = -Wl,-static -all-static
 
 # Libraries and plugin flags with which to link wireshark.
@@ -1030,7 +1030,7 @@ wireshark_LDADD = \
 	@LIBGNUTLS_LIBS@		\
 	@PORTAUDIO_LIBS@
 
-@ENABLE_STATIC_FALSE@tshark_LDFLAGS = -export-dynamic
+@ENABLE_STATIC_FALSE@tshark_LDFLAGS = -export-dynamic -pie
 @ENABLE_STATIC_TRUE@tshark_LDFLAGS = -Wl,-static -all-static
 
 # Libraries and plugin flags with which to link tshark.
@@ -1610,14 +1610,14 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/version_info.Po@am__quote@
 
 .c.o:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c $<
 
 .c.obj:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
diff -up wireshark-0.99.7-SVN-23507/gtk/Makefile.in.pie wireshark-0.99.7-SVN-23507/gtk/Makefile.in
--- wireshark-0.99.7-SVN-23507/gtk/Makefile.in.pie	2007-11-20 05:34:33.000000000 -0500
+++ wireshark-0.99.7-SVN-23507/gtk/Makefile.in	2007-11-20 15:01:33.000000000 -0500
@@ -979,14 +979,14 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/wsp_stat.Po@am__quote@
 
 .c.o:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c $<
 
 .c.obj:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
