Fix simple syntax error in python scripts

diff -up wireshark-1.4.0/epan/wspython/wspy_libws.py.orig wireshark-1.4.0/epan/wspython/wspy_libws.py
--- wireshark-1.4.0/epan/wspython/wspy_libws.py.orig	2010-08-30 00:17:30.000000000 +0200
+++ wireshark-1.4.0/epan/wspython/wspy_libws.py	2010-08-31 17:37:54.905966497 +0200
@@ -27,7 +27,6 @@ import platform
 
 __libwireshark = None
 
-/* XXX - We should probably return a full path here, at least on Windows. */
 def get_libws_libname():
   system = platform.system()
   if system == "Darwin":
