diff --git a/plugins/codecs/l16_mono/CMakeLists.txt b/plugins/codecs/l16_mono/CMakeLists.txt
index 99b3de3dcc..51589c37da 100644
--- a/plugins/codecs/l16_mono/CMakeLists.txt
+++ b/plugins/codecs/l16_mono/CMakeLists.txt
@@ -28,7 +28,7 @@ set_source_files_properties(
 )
 
 include_directories(
-	${CMAKE_SOURCE_DIR}/codec
+	${CMAKE_SOURCE_DIR}/codecs
 	${CMAKE_CURRENT_SOURCE_DIR}
 )
 
diff --git a/plugins/codecs/l16_mono/Makefile.am b/plugins/codecs/l16_mono/Makefile.am
index bbc2379eb3..0ad043cd44 100644
--- a/plugins/codecs/l16_mono/Makefile.am
+++ b/plugins/codecs/l16_mono/Makefile.am
@@ -26,7 +26,7 @@ NONGENERATED_REGISTER_C_FILES = \
 NONGENERATED_C_FILES = \
 	$(NONGENERATED_REGISTER_C_FILES)
 
-wiretap_plugin_LTLIBRARIES = l16mono.la
+codecs_plugin_LTLIBRARIES = l16mono.la
 
 l16mono_la_SOURCES = \
 	$(SRC_FILES)	\
