diff --git a/gtk/flow_graph.c b/gtk/flow_graph.c
index 73d7789..c10f381 100644
--- a/gtk/flow_graph.c
+++ b/gtk/flow_graph.c
@@ -464,6 +464,25 @@ flow_graph_on_ok                    (GtkButton       *button _U_,
 	}
 }
 
+static void
+flow_graph_on_cancel(GtkButton       *button _U_,
+		     gpointer         user_data)
+{
+	if (graph_analysis_data->dlg.window) {
+		window_destroy(graph_analysis_data->dlg.window);
+	}
+	window_destroy(GTK_WIDGET(user_data));
+}
+
+static gboolean
+flow_graph_on_delete(GtkButton       *button _U_,
+		     gpointer         user_data _U_)
+{
+	if (graph_analysis_data->dlg.window) {
+		window_destroy(graph_analysis_data->dlg.window);
+	}
+	return FALSE;
+}
 
 /****************************************************************************/
 /* INTERFACE                                                                */
@@ -618,9 +637,9 @@ flow_graph_dlg_create (void)
 	gtk_container_add (GTK_CONTAINER (hbuttonbox), bt_cancel);
 	GTK_WIDGET_SET_FLAGS(bt_cancel, GTK_CAN_DEFAULT);
 	gtk_tooltips_set_tip (tooltips, bt_cancel, "Cancel this dialog", NULL);
-	window_set_cancel_button(flow_graph_dlg_w, bt_cancel, window_cancel_button_cb);
+	g_signal_connect(bt_cancel, "clicked", G_CALLBACK(flow_graph_on_cancel), flow_graph_dlg_w);
 
-	g_signal_connect(flow_graph_dlg_w, "delete_event", G_CALLBACK(window_delete_event_cb), NULL);
+	g_signal_connect(flow_graph_dlg_w, "delete_event", G_CALLBACK(flow_graph_on_delete), NULL);
 	g_signal_connect(flow_graph_dlg_w, "destroy", G_CALLBACK(flow_graph_on_destroy), NULL);
 
 	gtk_widget_show_all(flow_graph_dlg_w);
