commit 7a47641e4b8b152264bb5d85807821a3d982eeae
Author: Radek Vokal <rvokal@fedoraproject.org>
Date:   Wed Jan 6 09:37:49 2016 +0100

    Add pkgconfig entry

diff --git a/Makefile.am b/Makefile.am
index 913e064..30f8107 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -57,6 +57,9 @@ EXTRA_PROGRAMS = wireshark-gtk wireshark tshark tfshark capinfos captype editcap
 dist_pkgdata_DATA = COPYING manuf services cfilters colorfilters dfilters \
 	smi_modules ipmap.html pdml2html.xsl
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = wireshark.pc
+
 #
 # Install global profiles in the "profiles" subdirectory
 #
diff --git a/wireshark.pc.in b/wireshark.pc.in
index 5889178..1e1b62f 100644
--- a/wireshark.pc.in
+++ b/wireshark.pc.in
@@ -6,9 +6,9 @@ sharedlibdir=${libdir}
 plugindir=@plugindir@
 
 Name: Wireshark
-Description: Network Packet Dissection Library
-Version: @VERSION@
+Description: Network Traffic Analyzer
+Version: @PACKAGE_VERSION@
 
-Requires: glib-2.0, gmodule-2.0, gthread-2.0
-Libs: -L${libdir} -lwireshark
-Cflags: -I${includedir}/wireshark
+Requires: Qt, glib-2.0, gmodule-2.0, gthread-2.0
+Libs: -L@libdir@ -lwireshark -lwiretap
+Cflags: -DWS_VAR_IMPORT=extern -DHAVE_STDARG_H -DWS_MSVC_NORETURN= -I@includedir@/wireshark -I@includedir@/wireshark/epan
