diff -up wireshark-1.6.8/epan/dissectors/packet-nfs.c.v4staus wireshark-1.6.8/epan/dissectors/packet-nfs.c
--- wireshark-1.6.8/epan/dissectors/packet-nfs.c.v4staus	2012-04-29 05:46:46.000000000 +0200
+++ wireshark-1.6.8/epan/dissectors/packet-nfs.c	2012-05-23 09:31:45.540759218 +0200
@@ -768,7 +768,7 @@ static int nfsv4_operation_tiers[] = {
 		 1 /* 50, NFS4_OP_LAYOUTGET */,
 		 1 /* 51, NFS4_OP_LAYOUTRETURN */,
 		 1 /* 52, NFS4_OP_SECINFO_NO_NAME */,
-		 1 /* 53, NFS4_OP_SEQUENCE */,
+		 4 /* 53, NFS4_OP_SEQUENCE */,
 		 1 /* 54, NFS4_OP_SET_SSV */,
 		 1 /* 55, NFS4_OP_TEST_STATEID */,
 		 1 /* 56, NFS4_OP_WANT_DELEGATION  */,
@@ -8977,6 +8977,8 @@ dissect_nfs_argop4(tvbuff_t *tvb, int of
 		ftree = proto_item_add_subtree(fitem, ett_nfs_argop4);
 	}
 
+	proto_item_append_text(tree, ", Ops(%d):", ops);
+
 	for (ops_counter=0; ops_counter<ops; ops_counter++)
 	{
 		op_summary[ops_counter].optext = g_string_new("");
@@ -9010,6 +9012,7 @@ dissect_nfs_argop4(tvbuff_t *tvb, int of
 		g_string_printf (op_summary[ops_counter].optext, "%s",
                                  val_to_str_ext_const(opcode, &names_nfsv4_operation_ext, "Unknown"));
 
+		proto_item_append_text(tree, " %s", opname);
 
 		switch(opcode)
 		{
@@ -9629,6 +9632,8 @@ dissect_nfs_resop4(tvbuff_t *tvb, int of
 		ftree = proto_item_add_subtree(fitem, ett_nfs_resop4);
 	}
 
+	proto_item_append_text(tree, ", Ops(%d):", ops);
+
 	for (ops_counter = 0; ops_counter < ops; ops_counter++)
 	{
 		op_summary[ops_counter].optext = g_string_new("");
@@ -9662,6 +9667,12 @@ dissect_nfs_resop4(tvbuff_t *tvb, int of
 		g_string_append_printf (op_summary[ops_counter].optext, "%s", opname);
 
 		offset = dissect_nfs_nfsstat4(tvb, offset, newftree, &status);
+		if (status != NFS4_OK) {
+			proto_item_append_text(tree, " %s(%s)", opname,
+				val_to_str_ext(status, &names_nfs_stat_ext, "Unknown error:%u"));
+		} else {
+			proto_item_append_text(tree, " %s", opname);
+		}
 
 		/*
 		 * With the exception of NFS4_OP_LOCK, NFS4_OP_LOCKT, and
