diff -up wireshark-1.0.2/gtk/Makefile.in.pie wireshark-1.0.2/gtk/Makefile.in
--- wireshark-1.0.2/gtk/Makefile.in.pie	2008-07-10 19:39:52.000000000 +0200
+++ wireshark-1.0.2/gtk/Makefile.in	2008-07-18 14:39:30.000000000 +0200
@@ -1009,14 +1009,14 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/wsp_stat.Po@am__quote@
 
 .c.o:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c $<
 
 .c.obj:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
diff -up wireshark-1.0.2/Makefile.in.pie wireshark-1.0.2/Makefile.in
--- wireshark-1.0.2/Makefile.in.pie	2008-07-10 19:39:55.000000000 +0200
+++ wireshark-1.0.2/Makefile.in	2008-07-18 14:39:30.000000000 +0200
@@ -1037,7 +1037,7 @@ noinst_HEADERS = \
 wireshark_optional_objects = @GETOPT_O@ @STRERROR_O@ \
 	@STRCASECMP_O@ @STRNCASECMP_O@ @MKSTEMP_O@ @STRPTIME_O@
 
-@ENABLE_STATIC_FALSE@wireshark_LDFLAGS = -export-dynamic
+@ENABLE_STATIC_FALSE@wireshark_LDFLAGS = -export-dynamic -pie
 @ENABLE_STATIC_TRUE@wireshark_LDFLAGS = -Wl,-static -all-static
 
 # Libraries and plugin flags with which to link wireshark.
@@ -1069,7 +1069,7 @@ wireshark_DEPENDENCIES = \
 	@INET_PTON_LO@			\
 	@INET_NTOP_LO@
 
-@ENABLE_STATIC_FALSE@tshark_LDFLAGS = -export-dynamic
+@ENABLE_STATIC_FALSE@tshark_LDFLAGS = -export-dynamic -pie
 @ENABLE_STATIC_TRUE@tshark_LDFLAGS = -Wl,-static -all-static
 
 # Libraries and plugin flags with which to link tshark.
@@ -1743,14 +1743,14 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/version_info.Po@am__quote@
 
 .c.o:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c $<
 
 .c.obj:
-@am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
+@am__fastdepCC_TRUE@	$(COMPILE) $(PIECFLAGS) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$<' object='$@' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
